File: frontend/src/types/platformDashboard.ts
export interface PlanMix {
  plan_code: string;
  plan_name: string;
  tenant_count: number;
}

export interface PlatformDashboard {
  active_tenant_count: number;
  trialing_count: number;
  churned_this_month_count: number;
  mrr_paise: number;
  plan_mix: PlanMix[];
  overdue_invoices_count: number;
}

File: backend/app/schemas/platform_dashboard.py
from pydantic import BaseModel


class PlanMixOut(BaseModel):
    plan_code: str
    plan_name: str
    tenant_count: int


class PlatformDashboardOut(BaseModel):
    active_tenant_count: int
    trialing_count: int
    churned_this_month_count: int
    mrr_paise: int
    plan_mix: list[PlanMixOut]
    overdue_invoices_count: int
